R CODE FINAL Alexander Shemetev

```diff
@@ -0,0 +1,791 @@
+
+
+#bankloans existing  - those observations for which there is a value for default (either 0 or 1)
+#bankloans_new - those observations for which default is NA
+
+
+#bankloans_existing <- subset(raw_data, raw_data$default == is.na(raw_data$default)) 
+#raw_data$def_exists = ifelse(raw_data$default > -1,1,0)
+#sum(raw_data$def_exists)
+bankloans_new <- subset(raw_data, is.na(raw_data$default))
+bankloans_existing <- subset(raw_data, raw_data$default > -1)
+
+# 7) Apply outlier_capping and process_missing on every column of bankloans_existing
+
+sapply(bankloans_existing, process_missing)
+# good
+sapply(bankloans_existing, outlier_capping)
+
+
+# 8) Calculate the correlation matrix with correlations represented by color (google heatmap, there is really simple solution using on
+#    of the packages mentioned in the course)
+matrix_raw <- as.matrix(raw_data)
+heatmap(matrix_raw)
+# Heatmap successfully created
+
+# 9) Create a 2x4 matrix of boxplots where x = default, y is all other columns (all 8 plots should be on one picture)
+# I did not quite clear understand the task. 
+# Here is simple boxplot for columns: 
+boxplot(raw_data, main = "boxplot.matrix(...., main = raw_data$default)",
+        notch = TRUE, col = 2:4)
+# Here is ggplot boxplot: 
+require(reshape2)
+library(ggplot2)
+library(gridExtra)
+library(MASS)
+# Compiled from official ggplot-2 documentation
+df <- subset(raw_data, select=c(default, age, ed, employ, address, income, debtinc, creddebt, othdebt))  
+p <- list(df)
+for (j in colnames(df)) {
+  p[[j]] <- ggplot(data=df, aes_string(x=df$default, y=j)) + # Specify dataset, input or grouping col name and Y
+    geom_boxplot(aes(fill=factor(type))) + guides(fill=FALSE) + # Boxplot by which factor + color guide
+    theme(axis.title.y = element_text(face="bold", size=14))  # Make the Y-axis labels bigger/bolder
+  
+}
+
+do.call(grid.arrange, c(p, ncol=2))
+
+
+# Compiled myself for resolving this task (METHOD 2)
+bp1 <- ggplot(raw_data, aes(x=default, y=age, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp1
+bp2 <- ggplot(raw_data, aes(x=default, y=ed, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp2 
+bp3 <- ggplot(raw_data, aes(x=default, y=employ, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp3 
+bp4 <- ggplot(raw_data, aes(x=default, y=address, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp4 
+bp5 <- ggplot(raw_data, aes(x=default, y=income, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp5 
+bp6 <- ggplot(raw_data, aes(x=default, y=debtinc, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp6 
+bp7 <- ggplot(raw_data, aes(x=default, y=creddebt, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp7 
+bp8 <- ggplot(raw_data, aes(x=default, y=othdebt, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp8 
+plist <- list(bp1, 	bp2, 	bp3, 	bp4, 	bp5, 	bp6, 	bp7, 	bp8)
+do.call("grid.arrange", c(plist))
+do.call(grid.arrange, c(plist, ncol=2))
+
```

Comment:
super long solution. works though. 

No need to set fill = default, as default has only two values, that is why color is not really good representation for the differences. You use color for continuous variables mostly